csi/node.go

```diff
@@ -180,22 +204,45 @@ func (s *OsdCsiServer) NodeUnpublishVolume(
 			err.Error())
 	}
 
-	// Verify target location is an existing directory
-	// See: https://github.com/container-storage-interface/spec/issues/60
-	if err = verifyTargetLocation(req.GetTargetPath()); err != nil {
-		return nil, status.Error(codes.NotFound, err.Error())
-	}
-
-	// Mount volume onto the path
-	if err = s.driver.Unmount(req.GetVolumeId(), req.GetTargetPath(), nil); err != nil {
+	// Get information about the target since the request does not
+	// tell us if it is for block or mount point.
+	// https://github.com/container-storage-interface/spec/issues/285
+	fileInfo, err := os.Lstat(req.GetTargetPath())
```

Comment:
I'm not sure if it would matter, because at this point we need to clean up. So if the driver was or not (and it passed check on line 100, then here we must clean up, right?